Detections/OktaSSO/FailedLoginsFromUnknownOrInvalidUser.yaml

```diff
@@ -21,14 +21,15 @@ query: |
   let FailureThreshold = 15;
   let FailedLogins = Okta_CL
   | where TimeGenerated > timeframe
-  | where eventType_s =~ "user.session.start" and outcome_reason_s =~ "UNKNOWN_USER"
-  | summarize count() by client_ipAddress_s, bin(TimeGenerated, 5m)
+  | where eventType_s =~ "user.session.start" and outcome_reason_s =~ "VERIFICATION_ERROR"
+  | summarize count() by actor_alternateId_s, client_ipAddress_s, bin(TimeGenerated, 5m)
   | where count_ > FailureThreshold
-  | project client_ipAddress_s;
+  | project client_ipAddress_s, actor_alternateId_s;
   Okta_CL
-  | where published_t > timeframe
-  | join kind=inner FailedLogins () on client_ipAddress_s
-  | where eventType_s =~ "user.session.start" and outcome_reason_s =~ "UNKNOWN_USER"
-  | summarize count() by ClientIP = client_ipAddress_s, City = client_geographicalContext_city_s, Country = client_geographicalContext_country_s, bin(TimeGenerated, 5m)
-  | sort by TimeGenerated desc
-  | extend timestamp = TimeGenerated, IPCustomEntity = ClientIP
+  | where TimeGenerated > timeframe
+  | join kind=inner FailedLogins () on client_ipAddress_s, actor_alternateId_s
```

Comment:
Might also be a good idea to put  FailedLogins in the bracket i.e. like below

| join kind=inner (FailedLogins) on client_ipAddress_s, actor_alternateId_s

Reply:
revisions submitted